File: 10_week_2/smarter_sum.py
sumArr = [0];

def smarter_sum(num):
	if num < 0:
		return None

	while len(sumArr) <= num:
		sumArr.append(sumArr[len(sumArr) - 1] + len(sumArr))

	return sumArr[num]

print(smarter_sum(3))

File: 10_week_2/smarter_sum.js
var sumArr = [0];

function smarter_sum(num) {
	if (num < 0) {
		return null;
	}
	while (sumArr.length <= num) {
		sumArr.push(sumArr[sumArr.length - 1] + sumArr.length);
	}
	return sumArr[num];
}

console.log(smarter_sum(3));
